Index: src/main/java/org/codehaus/mojo/gwt/shell/SuperDevModeMojo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/org/codehaus/mojo/gwt/shell/SuperDevModeMojo.java	(date 1508588214000)
+++ src/main/java/org/codehaus/mojo/gwt/shell/SuperDevModeMojo.java	(date 1522067172573)
@@ -129,6 +129,14 @@
     @Parameter(defaultValue = "${executedProject}")
     private MavenProject executedProject;
 
+    /**
+     * A directory containing GWT source to be prepended to the classpath for compiling.
+     * <p>
+     * Can be set from command line using '-Dgwt.sources=...'
+     */
+    @Parameter(property = "gwt.sources")
+    private String[] sources;
+
     @Override
     public void doExecute()
         throws MojoExecutionException, MojoFailureException
@@ -194,6 +202,14 @@
 
         try
         {
+            System.out.println("---x------------------x-----------------x----");
+            if(sources != null) {
+                for (String source : sources) {
+                    cmd.arg( "-src" ).arg(source);
+                    System.out.println("add source: " + source);
+                }
+            }
+            System.out.println(cmd);
             cmd.execute();
         }
         catch ( JavaCommandException e )
